launcher/ui/pages/modplatform/ModPage.cpp

```diff
@@ -241,6 +246,72 @@ void ModPage::onModSelected()
     ui->packView->adjustSize();
 }
 
+void ModPage::openUrl(const QUrl& url)
+{
+    // do not allow other url schemes for security reasons
+    if (!(url.scheme() == "http" || url.scheme() == "https")) {
+        qWarning() << "Unsupported scheme" << url.scheme();
+        return;
+    }
+
+    // detect mod URLs and search instead
+    static const QRegularExpression modrinth(QRegularExpression::anchoredPattern("(?:www\\.)?modrinth\\.com\\/mod\\/([^\\/]+)\\/?")),
+        curseForge(QRegularExpression::anchoredPattern("(?:www\\.)?curseforge\\.com\\/minecraft\\/mc-mods\\/([^\\/]+)\\/?")),
+        curseForgeOld(QRegularExpression::anchoredPattern("minecraft\\.curseforge\\.com\\/projects\\/([^\\/]+)\\/?"));
+
+    const QString address = url.host() + url.path();
+    QRegularExpressionMatch match;
+    const char* page;
+
+    if ((match = modrinth.match(address)).hasMatch())
+        page = "modrinth";
+    else if (APPLICATION->capabilities() & Application::SupportsFlame &&
+               ((match = curseForge.match(address)).hasMatch() || (match = curseForgeOld.match(address)).hasMatch()))
+        page = "curseforge";
```

Comment:
```suggestion
    match = modrinth.match(address)
    if (match.hasMatch()) {
        page = "modrinth";
    } else if (APPLICATION->capabilities() & Application::SupportsFlame) {
        match = curseForge.match(address);
        if (!match.hasMatch())
            match = curseForgeOld.match(address);

        if (match.hasMatch())
            page = "curseforge";
    }
```

i think this is a bit easier to read

Reply:
To me it looks like it's duplicating more code

Reply:
a bit of code duplication is fine if it helps keep the code readable. It's really not obvious to me what is going on in yours at first glance, especially with all those assignments inside the if statements

Reply:
Yes, I think I agree with you now.